src/backend/storeManagers/hyperplay/games.ts

```diff
@@ -472,13 +494,35 @@ async function resumeIfPaused(appName: string): Promise<boolean> {
   return isPaused
 }
 
+export async function cancelExtraction(appName: string) {
+  logInfo(
+    `cancelExtraction: Extraction will be canceled and downloaded zip will be removed`,
+    LogPrefix.HyperPlay
+  )
+
+  try {
+    process.noAsar = false
+
+    const extractZipService = inProgressExtractionsMap.get(appName)
+    if (extractZipService) {
+      extractZipService.cancel()
+    }
+  } catch (error: unknown) {
+    logInfo(
+      `cancelExtraction: Error while canceling the operation ${
+        (error as Error).message
+      } `,
+      LogPrefix.HyperPlay
+    )
+  }
+}
+
 export async function install(
   appName: string,
   { path: dirpath, platformToInstall, channelName, accessCode }: InstallArgs
 ): Promise<InstallResult> {
-  if (await resumeIfPaused(appName)) {
-    return { status: 'done' }
-  }
+  const wasPaused = await resumeIfPaused(appName)
```

Comment:
Could we not check if an extraction was in progress in `resumeIfPaused` and handle the resuming of the extraction there? I think it is confusing to have a wasPaused flag propagate throughout this entire method. It means we have to think through every line change through both flows, install and resume

Reply:
This was changed because upon resume download, and trying to continue the extraction does not happen anymore, I've tried various of methods, including calling within the `resumeIfPaused` with a lot of repeating logic, yet could not be executed. In this case, we had to change this method

Furthermore, this may change when we implement resume and pause for extraction

Reply:
Done, updated had to change quite of logic around